pkg/kubelet/dockershim/docker_service.go

```diff
@@ -271,7 +271,8 @@ func NewDockerService(config *ClientConfig, podSandboxImage string, streamingCon
 			cgroupDriver = dockerInfo.CgroupDriver
 		}
 		if len(kubeCgroupDriver) != 0 && kubeCgroupDriver != cgroupDriver {
-			return nil, fmt.Errorf("misconfiguration: kubelet cgroup driver: %q is different from docker cgroup driver: %q", kubeCgroupDriver, cgroupDriver)
+			// auto-detect and match the docker service cgroup driver in dockershim.
+			klog.Infof("The kubelet cgroup driver %q is different from the Docker cgroup driver %q. Automatically setting the kubelet cgroup driver to match it", kubeCgroupDriver, cgroupDriver)
```

Comment:
```suggestion
			klog.Infof("The kubelet cgroup driver %q is different from the Docker cgroup driver %q. Will use the Docker cgroup driver", kubeCgroupDriver, cgroupDriver)
```

It's pretty unclear to me which of the two we pick from this logline, although it's clear from the code below. I'd suggest we are explicit about which one we are using.

Reply:
I agree. @ehashman

Current one: `Automatically setting the kubelet cgroup driver to match it`
or
Make it more clear: `Automatically setting the kubelet cgroup driver to match the Docker setting`
or
ehashman's suggestion
` Will use the Docker cgroup driver`

@neolit123 I think ehashman's suggestion would be clear and more simplified. Do you think so?

Reply:
what @ehashman is proposing is clearer.